libs/core/algorithms/include/hpx/parallel/algorithms/sort_by_key.hpp

```diff
@@ -114,8 +113,7 @@ namespace hpx { namespace parallel { inline namespace v1 {
 
     template <typename ExPolicy, typename KeyIter, typename ValueIter,
         typename Compare = detail::less>
-    typename util::detail::algorithm_result<ExPolicy,
-        hpx::util::tagged_pair<tag::in1(KeyIter), tag::in2(ValueIter)>>::type
+    util::detail::algorithm_result_t<ExPolicy, std::pair<KeyIter, ValueIter>>
```

Comment:
Same here, a special type for the return value would be appropriate.

Reply:
Well, it's not a range algorithm, so using a `std::pair` is probably the right thing. I take back what I said.